### PR TITLE
feat(pattern/accordion) : add accordion - dev

### PR DESCRIPTION
**Pull request by hatem-benmansour-lmfr**
## I have read [the contributing guidelines](https://mozaic.adeo.cloud/Contributing/)

- [X] Yes
- [ ] No

## Does this PR introduce a [breaking change](https://mozaic.adeo.cloud/Contributing/Developers/GitConventions/#breaking-changes-)?

- [ ] Yes
- [X] No

## Describe the changes

GitHub issue number or Jira issue URL: N/A

### CODE REVIEW

File: packages/styles/components/_c.collapse.scss [hunk 1]
```diff
@@ -0,0 +1,64 @@
+.mc-collapse {
+  background: $color-grey-000;
```
Comment (gaelHugotLm) [inline packages/styles/components/_c.collapse.scss:2]: Hello, 

Ce serait mieux de mettre un `background-color` histoire de ne set que la couleur et non pas toutes les autres propriétés.

**Comment by tiloyi** [comment]
Deployment url: https://refspull568merge-dot-demo-dot-mozaic-design-system-dtrp.ey.r.appspot.com

File: packages/styles/components/_c.accordion.scss [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        max-height: 100vh;
+        padding-bottom: $mu150;
+      }
+    }
+
+    &:disabled {
+      ~ #{$parent}__label {
+        background-color: $color-grey-100;
+        background-image: url(inline-icons('control-more-24', $color-grey-600));
+        color: $color-grey-600;
```
Comment (tiloyi) [inline packages/styles/components/_c.accordion.scss:70]: Better to use accessible variable : 
~~color: $color-grey-600;~~ => `$color-font-light`

File: packages/styles/components/_c.accordion.scss [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    }
+
+    &:disabled:checked {
+      ~ #{$parent}__label {
+        background-image: url(inline-icons('control-less-24', $color-grey-600));
+      }
+      ~ #{$parent}__content {
+        max-height: 100vh;
+        padding-bottom: $mu150;
+      }
```
Comment (tiloyi) [inline packages/styles/components/_c.accordion.scss:86]: Not necessary. 
Already apply by the `&:checked ~ #{$parent}__content` selector in line 60.

File: src/docs/Components/Accordion/code.mdx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+Import *the settings* and *the accordion* scss files. 
+```scss
+@import 'settings-tools/_all-settings';
+
+@import 'components/_c.accordion';
+```
+
+## Basic usage
+To create a basic accordion, wrap a `checkbox`, a `label` and a `div` inside a `div` and apply the following classes:
```
Comment (tiloyi) [inline src/docs/Components/Accordion/code.mdx:16]: Can we rephrase it in a simpler way? Maybe : 

_To create a basic accordion,apply the following classes on this HTML elements:_

File: src/docs/Components/Accordion/code.mdx [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+      Fusce iaculis dolor nulla, a maximus ipsum sollicitudin et. Ut condimentum
+      at orci aliquam feugiat. Curabitur sagittis placerat leo sit amet
+      pharetra.
+    </p>
+  </div>
+</div>
+```
+
+<Preview path="basic" />
+
```
Comment (tiloyi) [inline src/docs/Components/Accordion/code.mdx:41]: Add an highlight to explain why we use inputs + labels. Say that it is an accordion that works without javascript.

File: src/docs/Components/Accordion/code.mdx [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+- `mc-accordion__trigger` on a input[checkbox] tag
+- `mc-accordion__label` on a label tag
+- `mc-accordion__content` on a div tag
+
+```html
+<div class="mc-accordion">
+  <input type="checkbox" id="basicUsage" class="mc-accordion__trigger" />
+  <label for="basicUsage" class="mc-accordion__label" aria-expand="false">
+    Accordion title
+    <span class="mc-accordion__icon"></span>
```
Comment (tiloyi) [inline src/docs/Components/Accordion/code.mdx:28]: I think this line is useless.

File: src/docs/Components/Accordion/code.mdx [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+
+Depending on the context of use, the accordion can have several states: 
+
+- Close
+- Open 
+- Close Disabled
+- Open Disabled
+
+<Preview path="open" />
+<Preview path="disable" />
```
Comment (tiloyi) [inline src/docs/Components/Accordion/code.mdx:52]: I think you can give examples of each of the 4 states in a single preview file.
Example => http://mozaic.adeo.cloud/previews/Foundations/Typography/HeadingStyles/previews/typographyHeading-sizes.html

File: src/docs/Components/Accordion/code.mdx [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+
+## Padding modifier
```
Comment (tiloyi) [inline src/docs/Components/Accordion/code.mdx:54]:
```
## Variations
### Padding modifier
```

File: packages/styles/components/_c.accordion.scss [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+    &:focus {
+      @include set-focus-styles-during();
+    }
+
+    &--no-padding {
+      padding-left: 0;
+    }
+  }
+
+  &__picto {
```
Comment (tiloyi) [inline packages/styles/components/_c.accordion.scss:36]: I think **picto** is a french word.
Maybe we can replace it with **icon**.

File: src/docs/Components/Accordion/code.mdx [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+You have the possibility to have an accordion with no padding.
+
+For this you have to add to the label the `mc-accordion__label--no-padding` modifier.
+
+<Preview path="no-padding" />
+
+## With icon
+You can add an icon into any accordions you want.
+
+You only need to add the `mc-accordion__picto` CSS class to you icon.
```
Comment (tiloyi) [inline src/docs/Components/Accordion/code.mdx:65]: I think **picto** is a french word.
Maybe we can replace it with **icon**.

Comment (tiloyi) [reply src/docs/Components/Accordion/code.mdx:65]: ~~to you icon~~ => to your icon

File: src/docs/Components/Accordion/code.mdx [hunk 11]
```diff
@@ -0,0 +1,98 @@
...
+        <path
+          d="M23 11l-5.59-5.59a2 2 0 00-2.82 0L9 11H5a2 2 0 00-2 2v12a2 2 0 002 2h22a2 2 0 002-2V13a2 2 0 00-2-2zm-3 14h-3.5v-5H20zm-4.5 0H12v-5h3.5zM27 25h-6v-5.5a.5.5 0 00-.5-.5h-9a.5.5 0 00-.5.5V25H5V13h4.83l.58-.59L16 6.83l5.59 5.58.58.59H27z"
+        ></path>
+        <path
+          d="M16.71 10.71a1 1 0 00-1.42 0l-3.58 3.58a1 1 0 00.7 1.71h7.18a1 1 0 00.7-1.71z"
+        ></path>
+      </symbol>
+    </svg>
+    Accordion title
+    <span class="mc-accordion__icon"></span>
```
Comment (tiloyi) [inline src/docs/Components/Accordion/code.mdx:86]: I think this line is useless.

File: src/docs/Components/Accordion/code.mdx [hunk 12]
```diff
@@ -0,0 +1,98 @@
...
+    <p>
+      Fusce iaculis dolor nulla, a maximus ipsum sollicitudin et. Ut condimentum
+      at orci aliquam feugiat. Curabitur sagittis placerat leo sit amet
+      pharetra.
+    </p>
+  </div>
+</div>
+```
+
+<Preview path="icon" />
```
Comment (tiloyi) [inline src/docs/Components/Accordion/code.mdx:98]: Add information about accessibility at the end of the file.

File: src/docs/Components/Accordion/code.mdx [hunk 13]
```diff
@@ -0,0 +1,98 @@
...
+
+## Padding modifier
+
+You have the possibility to have an accordion with no padding.
+
+For this you have to add to the label the `mc-accordion__label--no-padding` modifier.
+
+<Preview path="no-padding" />
+
+## With icon
```
Comment (tiloyi) [inline src/docs/Components/Accordion/code.mdx:62]:
```
### With icon
```

File: src/docs/Components/Accordion/previews/basic.preview.html [hunk 14]
```diff
@@ -0,0 +1,16 @@
+<div class="example">
+  <div class="mc-accordion">
+    <input type="checkbox" id="basic" class="mc-accordion__trigger" />
+    <label for="basic" class="mc-accordion__label" aria-expand="false">
+      Accordion title
+      <span class="mc-accordion__icon"></span>
```
Comment (tiloyi) [inline src/docs/Components/Accordion/previews/basic.preview.html:6]: I think this line is useless.

File: src/docs/Components/Accordion/previews/disable.preview.html [hunk 15]
```diff
@@ -0,0 +1,21 @@
...
+  <div class="mc-accordion">
+    <input
+      type="checkbox"
+      id="disabled"
+      class="mc-accordion__trigger"
+      disabled
+    />
+    <label for="disabled" class="mc-accordion__label" aria-expand="false">
+      Accordion title
+      <span class="mc-accordion__icon"></span>
```
Comment (tiloyi) [inline src/docs/Components/Accordion/previews/disable.preview.html:11]: I think this line is useless.

File: src/docs/Components/Accordion/previews/icon.preview.html [hunk 16]
```diff
@@ -0,0 +1,33 @@
...
+    <label
+      for="icon"
+      class="mc-accordion__label mc-accordion__label--with-picto"
+      aria-expand="false"
+    >
+      <svg class="ku-icon-32 mc-accordion__picto ">
+        <use href="#Store_StoreLM_32px"></use>
+      </svg>
+      Accordion title
+      <span class="mc-accordion__icon"></span>
```
Comment (tiloyi) [inline src/docs/Components/Accordion/previews/icon.preview.html:13]: I think this line is useless.

File: src/docs/Components/Accordion/previews/no-padding.preview.html [hunk 17]
```diff
@@ -0,0 +1,48 @@
+<div class="example">
+    <div class="mc-accordion">
+      <input type="checkbox" id="basic" class="mc-accordion__trigger" />
+      <label for="basic" class="mc-accordion__label mc-accordion__label--no-padding" aria-expand="false">
+        Accordion title
+        <span class="mc-accordion__icon"></span>
```
Comment (tiloyi) [inline src/docs/Components/Accordion/previews/no-padding.preview.html:6]: I think this line is useless.

File: src/docs/Components/Accordion/previews/open.preview.html [hunk 18]
```diff
@@ -0,0 +1,16 @@
+<div class="example">
+  <div class="mc-accordion">
+    <input type="checkbox" id="open" class="mc-accordion__trigger" checked />
+    <label for="open" class="mc-accordion__label" aria-expand="false">
+      Accordion title
+      <span class="mc-accordion__icon"></span>
```
Comment (tiloyi) [inline src/docs/Components/Accordion/previews/open.preview.html:6]: I think this line is useless.

File: src/docs/Components/Accordion/previews/stacked.preview.html [hunk 19]
```diff
@@ -0,0 +1,31 @@
+<div class="example">
+    <div class="mc-accordion">
+      <input type="checkbox" id="open" class="mc-accordion__trigger" />
+      <label for="open" class="mc-accordion__label" aria-expand="false">
+        Accordion title
+        <span class="mc-accordion__icon"></span>
```
Comment (tiloyi) [inline src/docs/Components/Accordion/previews/stacked.preview.html:6]: I think this line is useless.

File: yarn.lock [hunk 20]
```diff
@@ -10152,7 +10152,7 @@ is-boolean-object@^1.0.0:
   resolved "https://registry.npmjs.org/is-boolean-object/-/is-boolean-object-1.0.0.tgz#98f8b28030684219a95f375cfbd88ce3405dff93"
```
Comment (tiloyi) [inline yarn.lock:10152]: This file must not be present in the commits of this PR.

**Pull request merged**